src/main/java/org/cactoos/io/DigestOf.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.cactoos.Bytes;
+import org.cactoos.Input;
+
+/**
+ * Checksum calculation of {@link Input}.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.28
+ */
+public final class DigestOf implements Bytes {
+
+    /**
+     * The input.
+     */
+    private final Input source;
+
+    /**
+     * The buffer size.
+     */
+    private final int size;
+
+    /**
+     * The algorithm.
+     */
+    private final String algorithm;
+
+    /**
+     * Ctor.
+     * @param input The input
+     */
+    public DigestOf(final Input input) {
+        this(input, "SHA-256");
+    }
+
+    /**
+     * Ctor.
+     * @param input The input
+     * @param algorithm The algorithm
+     */
+    public DigestOf(final Input input, final String algorithm) {
```

Comment:
@fabriciofx the name `algorithm` is an attribute and an argument. Would be better to use different names.

Reply:
@yegor256 I'll fix it now. Thanks!